hicommonwealth/commonwealth
# Comment created Knock integration

## Link to Issue
Closes: #7684 

## Description of Changes
- Created the NotificaitonsProvider port which (for now) has only a single `triggerWorkflow` method.
- Creates `notifications.schemas.ts` which will contain schemas for all the data we send to the notifications provider. This can be used by product as a reference for the data that is available to them in the workflow builder.
- Created the `CommentCreated` event handler and attached it to the notifications policy on the Knock worker.

## Test Plan
- Add `CommentedCreated` to the `ALLOWED_EVENTS` env var in your `commonwealth/.env` file
- Set the `KNOCK_API_KEY` in your `.env` file (DM me to get the key)
- Run `yarn start-knock` in `packages/commonwealth/`
- In another terminal run `yarn start-message-relayer` in `packages/commonwealth`
- Start the Commonwealth app in another terminal
- Sign in on one account and create a thread (make sure you are subscribed to the thread -- refresh to double check)
  - note the user id of this account (`SELECT user_id FROM "Addresses" WHERE address = '[your address]';`)
- On another account, comment on the thread you created
- Now, join the common-dev Knock workspace (just create an account with your `@common.xyz` email)
- Go to https://dashboard.knock.app/common-dev/development/developers/logs/runs and look for a workflow with the recipient as the user id noted above and the workflow as 'comment-creation'
- Once you find the workflow run, click on it and then click on the `In-App` step and then click on the hyperlinked 'message' link. Once there, switch to the 'Preview' tab and make sure the comment appears as it should

## Deployment Plan


## Other Considerations

-
